karmada-scheduler panic when WeightPreference undefined

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->


**What happened**: When I tested the feature`SpreadByRegion`, I found out that `karmada-scheduler` crashed all the time and resource couldn't be scheduled. 

```
E0830 03:44:12.592298       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 185 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1a35440, 0x2eebb90})
	/src/Karmada/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x7d
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0004329a0})
	/src/Karmada/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x75
panic({0x1a35440, 0x2eebb90})
	/Go_1.17/src/runtime/panic.go:1038 +0x215
github.com/karmada-io/karmada/pkg/scheduler/core/spreadconstraint.isTopologyIgnored(...)
	/src/Karmada/pkg/scheduler/core/spreadconstraint/group_clusters.go:248
github.com/karmada-io/karmada/pkg/scheduler/core/spreadconstraint.GroupClustersWithScore({0xc00296ea80, 0x3, 0x3}, 0x5, 0xc0bb7fc3234c8bd8, 0x1d9b7f8)
	/src/Karmada/pkg/scheduler/core/spreadconstraint/group_clusters.go:74 +0xe1
github.com/karmada-io/karmada/pkg/scheduler/core.(*genericScheduler).selectClusters(0xc00296ea80, {0xc00296ea80, 0x3, 0x3}, 0x1f4bec8, 0xc00191b128)
	/src/Karmada/pkg/scheduler/core/generic_scheduler.go:143 +0x10e
github.com/karmada-io/karmada/pkg/scheduler/core.(*genericScheduler).Schedule(0xc000a56180, {0x1f4bec8, 0xc000056040}, 0xc00295d580, 0xc00191b128)
	/src/Karmada/pkg/scheduler/core/generic_scheduler.go:71 +0x3cc
github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).scheduleResourceBinding(0xc000624000, 0xc00191b010)
	/src/Karmada/pkg/scheduler/scheduler.go:424 +0x4a6
github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).doScheduleBinding(0xc000624000, {0xc0004311a0, 0x8}, {0xc0004311a9, 0x1b})
	/src/Karmada/pkg/scheduler/scheduler.go:327 +0x8a5
github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).doSchedule(0xc000a56120, {0xc0004311a0, 0x0})
	/src/Karmada/pkg/scheduler/scheduler.go:285 +0x49
github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).scheduleNext(0xc000624000)
	/src/Karmada/pkg/scheduler/scheduler.go:274 +0x105
github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).worker(...)
	/src/Karmada/pkg/scheduler/scheduler.go:198
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7f24c6aac238)
	/src/Karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00079c0c0, {0x1f14ec0, 0xc0004fe000}, 0x1, 0xc000562660)
	/src/Karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00079c0c0, 0x3b9aca00, 0x0, 0x0, 0xc000592fd0)
	/src/Karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0xaefea0, 0xc00033a140, 0xc000592fb8)
	/src/Karmada/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x25
created by github.com/karmada-io/karmada/pkg/scheduler.(*Scheduler).Run
	/src/Karmada/pkg/scheduler/scheduler.go:192 +0x21f
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x182a8a1]
``` 

The error code is in `pkg/scheduler/core/spreadconstraint/group_clusters.go` :

```go

func isTopologyIgnored(placement *policyv1alpha1.Placement) bool {
	strategy := placement.ReplicaScheduling
	spreadConstraints := placement.SpreadConstraints

	if len(spreadConstraints) == 0 || (len(spreadConstraints) == 1 && spreadConstraints[0].SpreadByField == policyv1alpha1.SpreadByFieldCluster) {
		return true
	}

	// If the replica division preference is 'static weighted', ignore the declaration specified by spread constraints.
	if strategy != nil && strategy.ReplicaSchedulingType == policyv1alpha1.ReplicaSchedulingTypeDivided &&
		strategy.ReplicaDivisionPreference == policyv1alpha1.ReplicaDivisionPreferenceWeighted &&
               // here: should add strategy.WeightPreference == nil 
		(len(strategy.WeightPreference.StaticWeightList) != 0 && strategy.WeightPreference.DynamicWeight == "") {
		return true
	}

	return false
}

```


**What you expected to happen**:  fix this bug.

**How to reproduce it (as minimally and precisely as possible)**: When configured the pp like this yaml:

```yaml
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: deploy
  namespace: test-test
spec:
  dependentOverrides:
    - deploy 
  placement:
    clusterAffinity:
      fieldSelector:
        matchExpressions:
        - key: provider 
          operator: In
          values: 
          - aws
    spreadConstraints:
    - spreadByField: region
      maxGroups: 2
      minGroups: 1 
    - spreadByField: cluster
      maxGroups: 3
      minGroups: 1
    replicaScheduling:
      replicaSchedulingType: Divided 
      replicaDivisionPreference: Weighted
  resourceSelectors: #optional, imutable
  - apiVersion: apps/v1
    kind: Deployment
    name: nginx-deployment
```

**Anything else we need to know?**:

**Environment**:
- Karmada version: 1.2.1
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

--- a/pkg/scheduler/core/spreadconstraint/group_clusters.go
+++ b/pkg/scheduler/core/spreadconstraint/group_clusters.go
@@ -245,7 +245,8 @@ func isTopologyIgnored(placement *policyv1alpha1.Placement) bool {
 	// If the replica division preference is 'static weighted', ignore the declaration specified by spread constraints.
 	if strategy != nil && strategy.ReplicaSchedulingType == policyv1alpha1.ReplicaSchedulingTypeDivided &&
 		strategy.ReplicaDivisionPreference == policyv1alpha1.ReplicaDivisionPreferenceWeighted &&
-		(len(strategy.WeightPreference.StaticWeightList) != 0 && strategy.WeightPreference.DynamicWeight == "") {
+		(strategy.WeightPreference == nil ||
+			len(strategy.WeightPreference.StaticWeightList) != 0 && strategy.WeightPreference.DynamicWeight == "") {
 		return true
 	}
 

